tests/chainer_tests/functions_tests/connection_tests/test_deconvolution_2d.py

```diff
@@ -87,54 +110,67 @@ def setUp(self):
             self.check_backward_options.update(atol=5e-4, rtol=5e-3)
             self.check_double_backward_options.update(atol=5e-3, rtol=5e-2)
 
-    @attr.gpu
-    def test_forward_consistency(self):
-        x_cpu = chainer.Variable(self.x)
-        W_cpu = chainer.Variable(self.W)
-        b_cpu = None if self.nobias else chainer.Variable(self.b)
-        with chainer.using_config('cudnn_deterministic',
-                                  self.cudnn_deterministic):
-            y_cpu = F.deconvolution_2d(
-                x_cpu, W_cpu, b_cpu, stride=self.stride, pad=self.pad,
+    def forward_cpu(self, inputs):
+        x, W, b = inputs
+        x_cpu = chainer.Variable(x)
+        W_cpu = chainer.Variable(W)
+        b_cpu = None if b is None else chainer.Variable(b)
+        y_cpu = F.deconvolution_2d(
+            x_cpu, W_cpu, b_cpu, stride=self.stride, pad=self.pad,
+            outsize=self.outsize)
+        return y_cpu,
+
+    def check_forward(self, inputs, backend_config):
+        y_expected, = self.forward_cpu(inputs)
+
+        if backend_config.use_cuda:
+            inputs = cuda.to_gpu(inputs)
+
+        x, W, b = inputs
+        x = chainer.Variable(x)
+        W = chainer.Variable(W)
+        b = None if b is None else chainer.Variable(b)
+
+        with backend_config:
+            y_actual = F.deconvolution_2d(
+                x, W, b, stride=self.stride, pad=self.pad,
                 outsize=self.outsize)
 
-        x_gpu = chainer.Variable(cuda.to_gpu(self.x))
-        W_gpu = chainer.Variable(cuda.to_gpu(self.W))
-        b_gpu = None if self.nobias else chainer.Variable(
-            cuda.to_gpu(self.b))
-        with chainer.using_config('use_cudnn', self.use_cudnn):
-            with chainer.using_config('cudnn_deterministic',
-                                      self.cudnn_deterministic):
-                with chainer.using_config('autotune', self.autotune):
-                    y_gpu = F.deconvolution_2d(
-                        x_gpu, W_gpu, b_gpu, stride=self.stride, pad=self.pad,
-                        outsize=self.outsize)
-
-        self.assertEqual(y_cpu.data.dtype, self.x_dtype)
-        self.assertEqual(y_gpu.data.dtype, self.x_dtype)
+        assert y_expected.data.dtype == self.x_dtype
+        assert y_actual.data.dtype == self.x_dtype
         testing.assert_allclose(
-            y_cpu.data, y_gpu.data.get(), **self.test_forward_options)
+            y_expected.data, y_actual.data.get(), **self.test_forward_options)
 
     @attr.gpu
-    def test_forward_consistency_im2col(self):
-        self.use_cudnn = 'never'
-        self.test_forward_consistency()
+    def test_forward(self, backend_config):
```

Comment:
~I am not sure about this but is this decorator preserved when the fixture is replaced by the backend injection?~ In any case, I think it can be removed since the `use_cuda` param in the injection ensures it.